app/src/app/user/profile/profile.ts

```diff
@@ -24,24 +24,35 @@ namespace app.user.profile {
                     onBoard: false
                 },
                 resolve: /*@ngInject*/{
-                    emailConfirmationToken:(
+                    emailConfirmed:(
                         userService:common.services.user.UserService,
                         $stateParams:IStateParams,
                         ngJwtAuthService:NgJwtAuth.NgJwtAuthService,
                         notificationService:common.services.notification.NotificationService,
-                        $location:ng.ILocationService
+                        $location:ng.ILocationService,
+                        $q:ng.IQService
                     ) => {
                         if(!_.isEmpty($stateParams.emailConfirmationToken)) {
-                            userService.confirmEmail(<common.models.User>ngJwtAuthService.getUser(), $stateParams.emailConfirmationToken)
+
+                            let emailToken = $stateParams.emailConfirmationToken;
+
+                            return userService.confirmEmail(<common.models.User>ngJwtAuthService.getUser(), emailToken)
                                 .then(() => {
-                                    notificationService.toast('Your email has successfully been updated').pop();
 
-                                    $location.search('emailConfirmationToken', null);
```

Comment:
Ah right, yes that makes sense. I think it would be best to just do a `$state.go` with reload as false on success and omit the `$stateParams.emailConfirmationToken` key (make sure to intersect the rest of the params thought).